interpreter/src/main/scala/pl/touk/nussknacker/engine/testmode/MockServiceCollectingListener.scala

```diff
@@ -0,0 +1,35 @@
+package pl.touk.nussknacker.engine.testmode
+
+import pl.touk.nussknacker.engine.api.component.Component
+import pl.touk.nussknacker.engine.api.process.WithCategories
+
+import scala.reflect.ClassTag
+
+case class MockComponentHolder(runId: TestRunId) extends Serializable {
+
+  def components[T <: Component: ClassTag]: Map[String, WithCategories[T]] = MockComponentsHolder.resultsForId[T](runId)
+
+  def clean(): Unit = MockComponentsHolder.cleanResult(runId)
+}
+
+
+object MockComponentsHolder {
+
+  val testRunId: TestRunId = TestRunId("mock")
```

Comment:
sth is wrong here :) this TestRunId should be generated for each test, so we can run them in parallel

Reply:
done